tezos/new_context/src/working_tree/working_tree.rs

```diff
@@ -90,43 +89,10 @@ enum Action {
     Remove(RemoveAction),
 }
 
-pub struct StagedCache {
```

Comment:
@sebastiencs not to be solved now (because 1) not a big deal for the in-mem store, 2) this StagedCache code was horrible), but what happens if after backtracking, I try to read a value that was already read before when working on the discarded tree? does it get read from the underlying storage again?

Reply:
Yes it will read from the underlying storage. Is this something we want to prevent ?

Reply:
Not important now, but something that could be improved in the future. It is a property that the old implementation had, but I don't really know how much it helped.